src/ast.c

```diff
@@ -245,20 +245,6 @@ static jl_value_t *scm_to_julia_(value_t e, int eo)
             }
             i64 = conv_to_int64(cp_data(cp), nt);
         }
-        if (
-#ifdef _P64
-            jl_compileropts.int_literals==32
-#else
-            jl_compileropts.int_literals!=64
-#endif
-            ) {
-            if (i64 > (int64_t)S32_MAX || i64 < (int64_t)S32_MIN)
-                return (jl_value_t*)jl_box_int64(i64);
-            return (jl_value_t*)jl_box_int32((int32_t)i64);
-        }
-        else {
-            return (jl_value_t*)jl_box_int64(i64);
-        }
```

Comment:
Whoa, deleted way too much here. All this code should remain, just with different platform ifdefs.